src/prommis/solvent_extraction/tests/test_sx_model_steady.py

```diff
@@ -1,284 +1,103 @@
-        m.fs.solex.mscontactor.aqueous_inlet_state[0].conc_mass_comp["Ca"].fix(109.542)
-        m.fs.solex.mscontactor.aqueous_inlet_state[0].conc_mass_comp["Fe"].fix(688.266)
-        m.fs.solex.mscontactor.aqueous_inlet_state[0].conc_mass_comp["Sc"].fix(0.032)
-        m.fs.solex.mscontactor.aqueous_inlet_state[0].conc_mass_comp["Y"].fix(0.124)
-        m.fs.solex.mscontactor.aqueous_inlet_state[0].conc_mass_comp["La"].fix(0.986)
-        m.fs.solex.mscontactor.aqueous_inlet_state[0].conc_mass_comp["Ce"].fix(2.277)
-        m.fs.solex.mscontactor.aqueous_inlet_state[0].conc_mass_comp["Pr"].fix(0.303)
-        m.fs.solex.mscontactor.aqueous_inlet_state[0].conc_mass_comp["Nd"].fix(0.946)
-        m.fs.solex.mscontactor.aqueous_inlet_state[0].conc_mass_comp["Sm"].fix(0.097)
-        m.fs.solex.mscontactor.aqueous_inlet_state[0].conc_mass_comp["Gd"].fix(0.2584)
-        m.fs.solex.mscontactor.aqueous_inlet_state[0].conc_mass_comp["Dy"].fix(0.047)
+    return m
 
-        m.fs.solex.mscontactor.aqueous_inlet_state[0].flow_vol.fix(62.01)
 
-        m.fs.solex.mscontactor.organic_inlet_state[0].conc_mass_comp["Al"].fix(1.267e-5)
-        m.fs.solex.mscontactor.organic_inlet_state[0].conc_mass_comp["Ca"].fix(2.684e-5)
-        m.fs.solex.mscontactor.organic_inlet_state[0].conc_mass_comp["Fe"].fix(2.873e-6)
-        m.fs.solex.mscontactor.organic_inlet_state[0].conc_mass_comp["Sc"].fix(1.734)
-        m.fs.solex.mscontactor.organic_inlet_state[0].conc_mass_comp["Y"].fix(2.179e-5)
-        m.fs.solex.mscontactor.organic_inlet_state[0].conc_mass_comp["La"].fix(0.000105)
-        m.fs.solex.mscontactor.organic_inlet_state[0].conc_mass_comp["Ce"].fix(0.00031)
-        m.fs.solex.mscontactor.organic_inlet_state[0].conc_mass_comp["Pr"].fix(3.711e-5)
-        m.fs.solex.mscontactor.organic_inlet_state[0].conc_mass_comp["Nd"].fix(0.000165)
-        m.fs.solex.mscontactor.organic_inlet_state[0].conc_mass_comp["Sm"].fix(1.701e-5)
-        m.fs.solex.mscontactor.organic_inlet_state[0].conc_mass_comp["Gd"].fix(3.357e-5)
-        m.fs.solex.mscontactor.organic_inlet_state[0].conc_mass_comp["Dy"].fix(8.008e-6)
+@pytest.mark.unit
+def test_structural_issues(model):
+    dt = DiagnosticsToolbox(model)
+    dt.report_structural_issues()
+    dt.assert_no_structural_warnings()
 
-        m.fs.solex.mscontactor.organic_inlet_state[0].flow_vol.fix(62.01)
 
-        return m
+@pytest.mark.component
+@pytest.mark.solver
+def test_solve(model):
 
-    @pytest.mark.component
-    def test_structural_issues(self, SolEx_frame):
-        model = SolEx_frame
-        dt = DiagnosticsToolbox(model)
-        dt.assert_no_structural_warnings()
+    initializer = SolventExtractionInitializer()
+    initializer.initialize(model.fs.solex)
 
-    @pytest.mark.component
-    def test_block_triangularization(self, SolEx_frame):
-        model = SolEx_frame
-        initializer = BlockTriangularizationInitializer(constraint_tolerance=1e-4)
+    try:
```

Comment:
On line 39 you are calling the initializer, and then immediately after you appear to be calling the initializer again in a try/except statement. That appears to be redundant.

Reply:
Yes I think I missed out that one, I have removed the try/except part